package.json

```diff
@@ -11,7 +11,9 @@
     "test-e2e-tokens": "LOG_LEVEL=error mocha --timeout 0 --bail --exit test/e2e/tokens/*.test.mjs ",
     "lint": "eslint . --ext js,mjs,jsx && find-unused-exports",
     "prepare": "husky install",
-    "doc:build:sdk": "jsdoc -c jsdoc.json cli/lib/nf3.mjs"
+    "doc:build:sdk": "jsdoc -c jsdoc.json cli/lib/nf3.mjs",
+    "build-adversary": "node test/adversary/transpile-adversary.mjs",
+    "adversary-test": "mocha --timeout 0 --bail --exit test/adversary.test.mjs"
```

Comment:
We should call `build-adversary` inside the `adversary-test` so that we don't have people run the test against "outdated" adversaries

Reply:
Adversary code is required at the stage of `./setup-nightfall` and `./start-nightfall`. Only after these two are completed do we run the tests. If there is no source code for adversary by `./start-nightfall `, it complains (don't remember if it fails, my memory says it does).

We could call `build-adversary` in the test. It would work if some adversary code already exists during build and start stage and this version of adversary is started. After this, `nodemon` will create the new code again during test and restart the adversary container. 

Considering we won't use `nodemon` in non dev deployments, I would recommend we get the developers used to calling `build-adversary` at the beginning of staging and prod deployments. Because they have to remember calling `./start-nightfall` with `-a`, as well as `./setup-nightfall` with `adversary` service too.

PS: We don't push `nightfall-adversary` to repo. It is part of `.gitignore` file